### PR TITLE
[FEAT]: Main Modal 

### PR DESCRIPTION
**Pull request by VipinDevelops**
## Issue(s) 
#25 

## Acceptance Criteria fulfillment
I added a GitHub Home Modal which is triggered by /gh or /github command and has buttons that can redirect users to different features/modals of the app

- [x] Add GitHub Home Modal which is triggered by /github or /gh command.
- [x]  GitHub Home Modal should have action buttons to trigger all the different features/modals in the app.
- [x] Conditional Rendering of action buttons that require authentication,

## Proposed changes (including videos or screenshots)
[Screencast from 2023-06-15 10-31-30.webm](https://github.com/RocketChat/Apps.Github22/assets/99081689/20c2d41d-d979-493b-b187-6d1539ad5201)

### CODE REVIEW

File: github/enum/Modals.ts [hunk 1]
```diff
@@ -38,116 +38,116 @@ export enum ModalsEnum {
...
+    ADD_SUBSCRIPTION_LABEL = 'Subscribe',
+    UPDATE_SUBSCRIPTION_ACTION = 'update-subscription',
+    UPDATE_SUBSCRIPTION_LABEL = 'Update',
+    DELETE_SUBSCIPTIONS_TITLE = 'Unsubscribe to Repositories',
+    ADD_SUBSCIPTIONS_TITLE = 'New Subscription',
+    OPEN_REPO_ACTION = 'open-repo',
+    OPEN_REPO_LABEL = 'Open',
+    REPO_NAME_INPUT = 'repo-name-input',
+    REPO_NAME_LABEL = 'Enter Full Repository Name',
+    REPO_NAME_PLACEHOLDER = 'Owner/Repository',
```
Comment (samad-yar-khan) [inline github/enum/Modals.ts:64]: @VipinDevelops  All these changes look unrelated to your PR. Looks like you are using a linter which is making changes as you save. This is bloating the PR. Please Fix this

File: github/modals/MainModal.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+    const user = slashcommandcontext?.getSender() || uikitcontext?.getInteractionData().user;
+
+    block.addContextBlock({
+        blockId: ModalsEnum.MAIN_MODAL_VIEW,
+        elements: [
+            block.newMarkdownTextObject(`Hey 👋 Welcome ${user?.username}`),
+        ],
+
+    });
```
Comment (samad-yar-khan) [inline github/modals/MainModal.ts:34]:
```
block.addContextBlock(
   {
        blockId: ModalsEnum.MAIN_MODAL_VIEW,
        elements: [
            block.newMarkdownTextObject(`Hey 👋 Welcome ${user?.username}`),
        ],
    }
);
```

File: github/modals/MainModal.ts [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        }),
+    })
+
+    block.addContextBlock({
+        blockId: ModalsEnum.MAIN_MODAL_VIEW,
+        elements: [
+            block.newMarkdownTextObject(ModalsEnum.MAIN_MODAL_GITHUB_SEARCH_DESCRIPTION),
+        ],
+    });
+
```
Comment (samad-yar-khan) [inline github/modals/MainModal.ts:57]:
```suggestion
```

File: github/modals/MainModal.ts [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    block.addContextBlock({
+        blockId: ModalsEnum.MAIN_MODAL_VIEW,
+        elements: [
+            block.newMarkdownTextObject(ModalsEnum.MAIN_MODAL_NEW_ISSUE_DESCRIPTION),
+        ],
+    });
+
+
```
Comment (samad-yar-khan) [inline github/modals/MainModal.ts:80]:
```suggestion
```

File: github/modals/MainModal.ts [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+            text: ModalsEnum.MAIN_MODAL_TITLE
+        },
+        close: block.newButtonElement({
+            text: {
+                type: TextObjectType.PLAINTEXT,
+                text: "Close",
+            }
+        }),
+        blocks: block.getBlocks(),
+
```
Comment (samad-yar-khan) [inline github/modals/MainModal.ts:141]:
```suggestion
```

File: github/modals/MainModal.ts [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+            actionId: ModalsEnum.TRIGGER_SUBSCRIPTIONS_MODAL,
+        }),
+    })
+    block.addContextBlock({
+        blockId: ModalsEnum.MAIN_MODAL_VIEW,
+        elements: [
+            block.newMarkdownTextObject(ModalsEnum.MAIN_MODAL_REPOSITORY_SUBSCRIPTIONS_DESCRIPTION),
+        ],
+    });
+
```
Comment (samad-yar-khan) [inline github/modals/MainModal.ts:101]: @VipinDevelops  There seems to be some inconsistency line spacing. Can you please fix these.

File: github/handlers/ExecuteBlockActionHandler.ts [hunk 7]
```diff
@@ -1010,40 +1014,83 @@ export class ExecuteBlockActionHandler {
...
+                            let index = -1;
+                            let currentIndex = 0;
+                            for (let issue of repoIssuesData.issue_list) {
+                                if (issue.issue_id == issueId) {
+                                    index = currentIndex;
+                                    break;
                                 }
-                                 let data = {
-                                    issues: repoIssuesData.issue_list,
-                                    pushRights : repoIssuesData.push_rights, //no access token, so user has no pushRights to the repo,
```
Comment (samad-yar-khan) [inline github/handlers/ExecuteBlockActionHandler.ts:1031]: @VipinDevelops  All these changes look unrelated to this PR. Can you please verify

**Review by samad-yar-khan — CHANGES_REQUESTED** [review]
@VipinDevelops  The PR seems to be bloated because of changes which seem unrelated to what this PR wishes to accomplish. Can you please update this.

File: github/enum/Modals.ts (hunk 1 above)
Comment (VipinDevelops) [reply github/enum/Modals.ts]: Sure I thought Its good to have formating in code , Removed :+1:

File: github/modals/MainModal.ts (hunk 2 above)
Comment (VipinDevelops) [reply github/modals/MainModal.ts:34]: :+1:

File: github/handlers/ExecuteBlockActionHandler.ts (hunk 7 above)
Comment (VipinDevelops) [reply github/handlers/ExecuteBlockActionHandler.ts]: Yeah I removed all the formatting changes please have a look :smile:

**Review by samad-yar-khan — CHANGES_REQUESTED** [review]

**Comment by VipinDevelops** [comment]
Hello @samad-yar-khan Sir I fixed the merge conflict and made other changes and tested the PR

**Review by samad-yar-khan — APPROVED** [review]
@VipinDevelops  Looks good to me ! Thankyou. Can you please open another PR to updated the ReadMe with this change. We have added all commands there.
